services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -287,7 +291,7 @@ public RouterServer(
       List<ServiceDiscoveryAnnouncer> serviceDiscoveryAnnouncers,
       Optional<DynamicAccessController> accessController,
       Optional<SSLFactory> sslFactory,
-      MetricsRepository metricsRepository) {
+      VeniceMetricsRepository metricsRepository) {
```

Comment:
Modified it to use `MetricsRepository` all over the code and when checking `instanceof` in `RouterHttpRequestStats` to setup opentelemetry metrics there.
Why this route: This will not be a breaking change anymore. We can break it together for all components with a later effort.